### PR TITLE
fix_2118

### PR DESCRIPTION
**Pull request by NiceYouKnow**
Remove the fuzzy regularity of the 942310 rule。

### CODE REVIEW

**Comment by NiceYouKnow** [comment]
@franbuehler , #2118 already resolved, please review.

File: tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942310.yaml [hunk 1]
```diff
@@ -17,7 +17,40 @@
             Host: localhost
           method: GET
           port: 80
+          #This is a payload with the fuzzy regularity removed. Now, it is a white sample. regex: [\"'`]\s+and\s*?=\W
```
Comment (theseion) [inline tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942310.yaml:20]: Move this to the description:
```yaml
desc: |
    Chained SQL injection attempts 2/2.

    This test originally checked the expression `[\"'`]\s+and\s*?=\W`. We opted to remove that expression as it 
    does not appear to match anything useful (https://github.com/coreruleset/coreruleset/issues/2118).
    This test now checks that no match occurs for something that would have matched the original expression.
```

File: tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942310.yaml [hunk 2]
```diff
@@ -17,7 +17,40 @@
...
+            Host: localhost
+          method: GET
+          port: 80
+          uri: "/?var='%20and%201%20in%20(select%20min(name)%20from%20sysobjects%20where%20xtype%20%3D%20'U'%20and%20name%20%3E%20'.')%20--"
+          version: HTTP/1.0
+        output:
+          log_contains: id "942310"
+  -
+    test_title: 942310-3
+    desc: "chained SQL injection attempts 2/2"
```
Comment (theseion) [inline tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942310.yaml:43]: Please describe the regular expression this test is designed to check.

File: tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942310.yaml [hunk 3]
```diff
@@ -17,7 +17,40 @@
...
           method: GET
           port: 80
+          #This is a payload with the fuzzy regularity removed. Now, it is a white sample. regex: [\"'`]\s+and\s*?=\W
           uri: "/?var=%22%27%20and%20%3d%20bar"
           version: HTTP/1.0
+        output:
+          no_log_contains: id "942310"
+  -
+    test_title: 942310-2
+    desc: "chained SQL injection attempts 2/2"
```
Comment (theseion) [inline tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942310.yaml:27]: Please describe the regular expression this test is designed to check.

**Review by theseion — CHANGES_REQUESTED** [review]
Thanks! Just needs a bit more documentation.

**Comment by theseion** [comment]
@NiceYouKnow You can use the `desc` property for describing the test, no need for comments (`#`). Use `data: |` for multiline descriptions, as I've done in the example.

**Comment by dune73** [comment]
Thanks @theseion.

@NiceYouKnow, we have talked about your welcome PR in our dev meeting earlier this week and we assigned the review to @theseion, since @franbuehler is a bit overburdened with many other issues / PRs / reviews. I hope you can follow the suggestion of @theseion.

-> (Results of the August project meeting](https://github.com/coreruleset/coreruleset/issues/2185#issuecomment-913804183)

**Comment by dune73** [comment]
Thank you @NiceYouKnow.

File: tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942310.yaml [hunk 4]
```diff
@@ -1,23 +1,57 @@
...
+            method: GET
+            port: 80
+            uri: "/?var=%22%27%20and%20%3d%20bar"
+            version: HTTP/1.0
+          output:
+            no_log_contains: id "942310"
+  - test_title: 942310-2
+    desc: |
+      Chained SQL injection attempts 2/2.
+      This is a regular test, this is a valid payload: \(\s*?select\s*?\w+\s*?\(
```
Comment (theseion) [inline tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942310.yaml:29]: Pleas change this to:

```suggestion
This test checks for a positive match of `\(\s*?select\s*?\w+\s*?\(`.
```

File: tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942310.yaml [hunk 5]
```diff
@@ -1,23 +1,57 @@
...
+            method: GET
+            port: 80
+            uri: "/?var='%20and%201%20in%20(select%20min(name)%20from%20sysobjects%20where%20xtype%20%3D%20'U'%20and%20name%20%3E%20'.')%20--"
+            version: HTTP/1.0
+          output:
+            log_contains: id "942310"
+  - test_title: 942310-3
+    desc: |
+      Chained SQL injection attempts 2/2.
+      This is a regular test, this is a valid payload: order\s+by\s+if\w*?\s*?\(
```
Comment (theseion) [inline tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942310.yaml:45]: Please change this to:
```suggestion
      This test checks for a positive match of `order\s+by\s+if\w*?\s*?\(`.
```

**Review by theseion — CHANGES_REQUESTED** [review]
Looks good to me. Please update the descriptions as proposed and we are good to go.

**Review by theseion — APPROVED** [review]

**Comment by theseion** [comment]
Thanks again!

**Pull request merged**